diff --git a/src/api/environment.cc b/src/api/environment.cc
index 3c5665a2d2..f314700344 100644
--- a/src/api/environment.cc
+++ b/src/api/environment.cc
@@ -293,6 +293,12 @@ Environment* CreateEnvironment(IsolateData* isolate_data,
                                       Environment::kOwnsProcessState |
                                       Environment::kOwnsInspector));
   env->InitializeLibuv(per_process::v8_is_profiling);
+
+
+#if HAVE_INSPECTOR && NODE_USE_V8_PLATFORM
+  env->InitializeInspector(nullptr);
+#endif
+
   if (env->RunBootstrapping().IsEmpty()) {
     return nullptr;
   }
